tapir/wirgarten/forms/registration/chicken_shares.py

```diff
@@ -0,0 +1,125 @@
+<li>Der Vertrag über den Hühneranteil verlängert sich automatisch um ein weiteres Jahr, wenn der Gemüse-Erntevertrag nicht gekündigt wird und der Hühneranteil nicht separat schriftlich drei Monate vor Laufzeitende per Mail an <a href="mailto:{site_email}">{site_email}</a> gekündigt wurde.</li>
+<li>Der Vertrag über einen Hühneranteil beinhaltet ein Probequartal. Innerhalb der ersten drei Monate der Vertragslaufzeit kannst du zum Ende des dritten Monats per Mail an <a href="mailto:{site_email}">{site_email}</a> kündigen.</li> 
+<li>Der Hühneranteil kann mit Zustimmung von der {site_name} an Dritte übertragen, erhöht oder verringert werden. Dafür muss vorher eine Mail an <a href="mailto:{site_email}">{site_email}</a> geschrieben werden.</li>
+<li>So du eine Rechnung erhalten möchtest, schreib uns eine Mail an <a href="mailto:{site_email}">{site_email}</a></li>
+</ol>
+</div>
+<div><img style="max-height:24em; margin-left:2em" src="/static/wirgarten/images/chicken_share.webp"/></div>
+</div>
+""".format(
+        site_email=SITE_EMAIL, site_name=SITE_NAME
+    )
+
+    def summary(self):
+        return """<span style="float:right; font-size:1.25rem">= <strong><span id="additional_shares_total">0.00</span> €</strong> / Monat<script>
+                  const calculatePrice = (chicken_share) => {{
+                        const [key, price] = chicken_share.split(':');
+                        return document.getElementsByName('Additional Shares-' + key)[0].value * price;
+                  }};
+
+                   const resultElem = document.getElementById('additional_shares_total');
+
+                   const calculateTotal = () => {{
+                        return '{chicken_shares}'.split(',').map(calculatePrice).reduce((a,b) => a + b);
+                   }}
+
+                  const handleChange = (event) => {{
+                        if(event){{
+                            if(event.target.value > {max_shares}){{
+                                event.target.value = {max_shares}
+                            }} else if(event.target.value < 0){{
+                                event.target.value = 0
+                            }}
+                        }}
+
+                        resultElem.innerText = calculateTotal().toFixed(2);
+                  }}
+
+                  '{chicken_shares}'.split(',').forEach(chicken_share => {{
+                    const key = chicken_share.split(":")[0];
+                    const input = document.getElementsByName('Additional Shares-' + key)[0];
+                     input.max = {max_shares};
+                    input.addEventListener('change', handleChange);
+                  }});
+
+                  handleChange();
+                   </script></span>""".format(
```

Comment:
I would put this in a typescript file and import that.

Also, we probably want the price calculation to be done on the python side, accessible with something like an API call, to make sure that we get consistent results (assuming that we use those calculations somewhere else, to generate receipts maybe)

Reply:
The calculation for receipts etc will be done in Python (or in a query) based on which active products a user has selected. 

I understand your point about consistency, but I don't really like the idea of doing a simple multiplication via API. 
It wil probably make the form feel laggy..

I can definitely generify a lot of the summary code for the first two forms though and it is probably easy to write tests for this.